cloud/gcp/pubsub/subscription/detectors-subscription.tf

```diff
@@ -0,0 +1,96 @@
+		detect(when(signal >= ${var.push_latency_threshold_warning}) and when(signal < ${var.push_latency_threshold_critical})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high >= ${var.push_latency_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.push_latency_disabled_critical, var.push_latency_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.push_latency_notifications_critical, var.push_latency_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high >= ${var.push_latency_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.push_latency_disabled_warning, var.push_latency_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.push_latency_notifications_warning, var.push_latency_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "push_latency_anomaly" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] GCP PubSub subscription latency on push endpoint changed abnormally"
+
+	program_text = <<-EOF
+		from signalfx.detectors.against_recent import against_recent
+		signal = data('subscription/push_request_latencies', filter=filter('monitored_resource', 'pubsub_subscription') and ${module.filter-tags.filter_custom})${var.push_latency_anomaly_aggregation_function}.${var.push_latency_anomaly_transformation_function}(over='${var.push_latency_anomaly_transformation_window}').publish('signal')
+		against_recent.detector_mean_std(signal, current_window=duration('${var.push_latency_anomaly_current_window}'), historical_window=duration('${var.push_latency_anomaly_historical_window}'), fire_num_stddev=${var.push_latency_anomaly_fire_num_stddev}, clear_num_stddev=${var.push_latency_anomaly_clear_num_stddev}, orientation='${var.push_latency_anomaly_orientation}', calculation_mode='${var.push_latency_anomaly_calculation_mode}').publish('CRIT')
+	EOF
+
+	rule {
+		description           = "at >= ${var.push_latency_anomaly_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.push_latency_anomaly_disabled_critical, var.push_latency_anomaly_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.push_latency_anomaly_notifications_critical, var.push_latency_anomaly_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
```

Comment:
What do you think about adding more detectors, as mentioned in [google docs](https://cloud.google.com/pubsub/docs/monitoring)?

Some detectors for:

* `subscription/num_undelivered_messages`
* `subscription/pull_ack_message_operation_count`, filtered by `response_code != "success"`
* `subscription/push_request_count`, filtered by `response_code != "success"`
* `subscription/push_request_count`
* `subscription/num_outstanding_messages`
* `subscription/num_undelivered_messages`

I don't know if it's too much...

Reply:
yes you are right. that said I think we should merge the PR with the current scope which works and add them in another PR. for now the priority is to translate "smartly" everything from https://github.com/claranet/terraform-signalfx-detectors.

I added an issue this in our bugtracker to not forget.